### PR TITLE
M365 Defender integration - fix on process.command_line 

### PR DESCRIPTION
**Pull request by daverick**
`process.command_line` was overwritten by `process.parent.command_line` due to a typo in the script parsing the parent.command line 

## Proposed commit message

fix process.command_line overwritten by process.parent.command_line

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

**Review by kcreddy — COMMENTED** [review]
Can you add an entry into `changelog.yml` like this: https://github.com/elastic/integrations/blob/main/packages/m365_defender/changelog.yml#L12-L16. Since this is a bugfix, the version here would be `2.5.1`

Can you also update `manifest.yml` [here](https://github.com/elastic/integrations/blob/main/packages/m365_defender/manifest.yml#L4) with version `2.5.1`?

**Comment by daverick** [comment]
@kcreddy done.
I have update the tests too. I don't have a way to really test these changes are corrects ;-(

File: packages/m365_defender/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.5.1"
+  changes:
+    - description: fix process.command_line overwritten by process.parent.command_line
+    type: bug
+    link: https://github.com/elastic/integrations/pull/8872
```
Comment (kcreddy) [inline packages/m365_defender/changelog.yml:6]: The CI seems to be failing here due to formatting 
`
Error: checking package failed: formatting the integration failed (path: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1705066585826701134/elastic/integrations/packages/m365_defender, failFast: false): walking through the integration files failed: formatting file failed (path: /opt/buildkite-agent/builds/bk-agent-prod-gcp-1705066585826701134/elastic/integrations/packages/m365_defender/changelog.yml): formatting file content failed: unmarshalling YAML file failed: yaml: line 3: did not find expected '-' indicator
`

```suggestion
- version: "2.5.1"
  changes:
    - description: fix process.command_line overwritten by process.parent.command_line
      type: bugfix
      link: https://github.com/elastic/integrations/pull/8872
```
Here's the suggested change to format it correctly. You can run the command ` elastic-package build && elastic-package format && elastic-package lint && elastic-package check` and ensure this passes after the change.

**Comment by kcreddy** [comment]
>  I don't have a way to really test these changes are corrects ;-(

After the change to the painless script, your change is valid according to this block of code:

https://github.com/elastic/integrations/pull/8872/files#diff-9a137094a48bc06c8c31eb412ef82a2d3bd2d2667d57e30dff9f299bbf5870c5L990-R1013

**Pull request merged**